### ISSUE TITLE
Repeated field names in union are accepted

### ISSUE THREAD

**Issue by jonjove** [stellar/xdrgen#133]
### What problem does your feature solve?

```
union Test switch (int v)
{
case 0:
     int x;
case 1:
     int x;
}
```

compiles successfully but breaks xdrc due to the repeated `x`. The fix for a real example that broke stellar-core can be seen in https://github.com/stellar/stellar-xdr-next/pull/20.

### What would you like to see?

This shouldn't compile.

### What alternatives are there?

TODO

**Comment by jonjove** [comment]
This is specifically mentioned in [RFC-4506 section 6.4 (4)](https://datatracker.ietf.org/doc/html/rfc4506#section-6.4)

> Similarly, variable names must be unique within the scope of struct and union declarations.  Nested struct and union declarations create new scopes.

**Comment by leighmcculloch** [comment]
tl;dr: I don't think it is worth making it not compile, as it has no material impact on the generated Rust code. Xdrgen is not focused on being an exact-perfect implementation of XDR, but a practical / product one. There are also a many other ways that xdrgen and xdrc differ in what they allow and support (non-option circular refs, enum variants, union default arms, and others), so making this one case not compile would not holistically address the real issue.

>compiles successfully but breaks xdrc due to the repeated x

The real issue of why this is inconvenient is that we make changes to the XDR without checking if the change compiles across all the generators we use in the Stellar ecosystem. It might be worth addressing that.
